docs/components/table-data.md

```diff
@@ -0,0 +1,479 @@
+
+The identifier should be a string and will default to `''` if not provided. In that scenario, we will generate a random ID for the identifier every time the table is mounted.
+
+:::danger DANGER
+This identifier must be unique across all KTableData instances across the entire Vue app, otherwise there is a risk that [SWRV](https://docs-swrv.netlify.app/) will return the cached data of the wrong table.
+:::
+
+### fetcherCacheKey
+
+Whenever the cache key is changed the fetcher will automatically be called and attempt to fetch new table data.
+
+```vue
+<template>
+  <KTableData
+    cache-identifier="fetcher-cache-key-example-table"
+    :fetcher-cache-key="String(cacheKey)"
+    :fetcher="fetcher"
+    :headers="headers"
+  />
+</template>
+
+<script setup lang="ts">
+const cacheKey = ref<number>(1)
+
+const itemDeleted = (): void => {
+  cacheKey.value++ // triggers refetch or reactivity
+}
+</script>
+```
+
+An alternate implementation is to apply a `key` attribute to the KTableData in conjunction with a `fetcher` and the [SWRV](https://docs-swrv.netlify.app/) `revalidate` function. To prevent unnecessary calls on mount, the `key` `ref` should have an initial value of `0`.
+
+Since the `fetcher` function will handle the initial fetch of the data, we want the cache key for the `revalidate` function to be a falsey value on page load (to avoid an unnecessary duplicate call), and will manually call `revalidate()` and increment the `key` to trigger a refetch and redraw of the table.
+
+```html
+<template>
+  <KTableData :key="key" :fetcher="fetcher" :headers="headers" />
+</template>
+
+<script setup lang="ts">
+const key = ref<string>(0) // initialized to zero
+
+const fetcher = async ({ pageSize, page, query, offset = null }) => {
+  try {
+    const res = await services.getAll(pageSize, offset)
+
+    // handle data
+  } catch (error) {
+    // handle error
+  }
+}
+
+const { revalidate } = composables.useRequest(
+  () => key.value && `service-list-${key.value}`, // will evaluate to falsey on mount, preventing an extra call
+  () => { return fetcher() }
+)
+
+const handleDelete = (id): void => {
+  try {
+    const res = await services.delete(id)
+
+    key.value++
+    revalidate()
+  } catch (error) {
+    // handle error
+  }
+}
+</script>
+```
+
+### searchInput
+
+Pass in a string of search query for server-side table filtering.
+
+### sortable
+
+Set this to `false` to disable ability to sort.
+
+### hidePaginationWhenOptional
+
+Set this to `true` to hide pagination when the table record count is less than or equal to the page size. Defaults to `false`.
+
+### hideToolbar
+
+Prop for hiding toolbar. Useful when elements provided in the toolbar are not actionable (for example, when toolbar contains filter controls, however the fetcher returns no results as there are no records to filter by).
+
+## Events
+
+### state
```

Comment:
This event is named the same as it was in KTable, but an event name of "state" is very broad and doesn't actually return the "state" of a large portion of the table's configuration/data